### DISCUSSION TITLE
When is support for head_errlog and zilsaxattr features coming to a stable release?

### DISCUSSION THREAD

**Discussion by meithan** [openzfs/zfs#14670] [Q&A] [answered]
I recently had to switch to the ZFS master branch temporarily to fix an issue, but inadvertently did a `zpool upgrade`, enabling the `head_errlog` and `zilsaxattr` features on my pools and making them incompatible with the latest stable (2.1.9) and previous releases.

I had a look at the zfs-2.1.10-staging branch and don't see references to these features yet, but I understand the branch might still not be near its release state.

Is there an estimated time frame for these features (and perhaps also `block_cloning`) to be integrated into the next stable release? Thanks!

**Comment by rincebrain** [comment, marked as answer]
Point releases (usually) don't get new feature flags, that would be when a new stable branch is cut, and even if 2.2 or whatever it's going to be called were branched tomorrow, it's going to be a fair while of stabilizing and testing before it turns into a 2.2.0 release.

**Reply by meithan** [reply]
Thanks a lot for the answer.

Then I guess I've put myself into a corner by doing things rushedly without properly understanding the consequences. My best course of action (to recover compatibility with the stable release) is to create new zpools on a system running the latest stable release and transfer the data over to them, I suppose?

**Reply by rincebrain** [reply]
You can create pools that are compatible with older versions on newer ones, if you specify things carefully (or, on 2.1 and up, you can use -o compatibility= to use preset compatibility option lists), but yes, making new pools compatible with older systems and hauling your data around would be good.
